services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/BeanTableSchema.java

```diff
@@ -41,72 +40,61 @@
 import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbIgnore;
-import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbImmutable;
 
 /**
  * Implementation of {@link TableSchema} that builds a table schema based on properties and annotations of a bean
  * class. Example:
  * <pre>
  * <code>
  * {@literal @}DynamoDbBean
- * public class CustomerAccount {
- *     private String unencryptedBillingKey;
+ * public class Customer {
+ *     private String accountId;
+ *     private int subId;            // primitive types are supported
+ *     private String name;
+ *     private Instant createdDate;
  *
  *     {@literal @}DynamoDbPartitionKey
- *     {@literal @}DynamoDbSecondarySortKey(indexName = "accounts_by_customer")
- *     public String accountId;
+ *     public String getAccountId() { return this.accountId; }
+ *     public void setAccountId(String accountId) { this.accountId = accountId; }
  *
  *     {@literal @}DynamoDbSortKey
- *     {@literal @}DynamoDbSecondaryPartitionKey(indexName = "accounts_by_customer")
- *     public String customerId;
- *
- *     {@literal @}DynamoDbAttribute("account_status")
- *     public CustomerAccountStatus status;
- *
- *     {@literal @}DynamoDbFlatten(dynamoDbBeanClass = Customer.class)
- *     public Customer customer;
- *
- *     public Instant createdOn;
+ *     public int getSubId() { return this.subId; }
+ *     public void setSubId(int subId) { this.subId = subId; }
  *
- *     // All public fields must be opted out to not participate in mapping
- *     {@literal @}DynamoDbIgnore
- *     public String internalKey;
+ *     // Defines a GSI (customers_by_name) with a partition key of 'name'
+ *     {@literal @}DynamoDbSecondaryPartitionKey(indexNames = "customers_by_name")
+ *     public String getName() { return this.name; }
+ *     public void setName(String name) { this.name = name; }
  *
- *     public enum CustomerAccountStatus {
- *         ACTIVE,
- *         CLOSED
- *     }
+ *     // Defines an LSI (customers_by_date) with a sort key of 'createdDate' and also declares the
+ *     // same attribute as a sort key for the GSI named 'customers_by_name'
+ *     {@literal @}DynamoDbSecondarySortKey(indexNames = {"customers_by_date", "customers_by_name"})
+ *     public Instant getCreatedDate() { return this.createdDate; }
+ *     public void setCreatedDate(Instant createdDate) { this.createdDate = createdDate; }
  * }
- * </code>
- * {@literal @}DynamoDbBean
- * public class Customer {
- *     public String name;
  *
- *     {@literal public List<String> address;}
- * }
- * }
  * </pre>
+ *
  * @param <T> The type of object that this {@link TableSchema} maps to.
  */
 @SdkPublicApi
-public final class BeanTableSchema<T> implements TableSchema<T> {
+public final class BeanTableSchema<T> extends WrappedTableSchema<T, StaticTableSchema<T>> {
```

Comment:
I am thinking I should add 'implements TableSchema<T>' here because WrappedTableSchema is an internal class and I want the TableSchema interface to be part of the public contract of this class.